### ISSUE TITLE
Where to put common RIB's code?

### ISSUE THREAD

**Issue by kronik** [uber/RIBs#201]
I'm struggling with a problem with base RIB code. Let's say we have a RIB with photo grid. When you tap on a photo it opens this photo in full screen or in an editor. Then we want to add another RIB with some filters. So it has additional functions in the interactor, presenter, and router. Builder and view controller is also different. There may be other RIBs with this base/common functionality. Any ideas on how to solve this issue?

**Comment by AttwellBrian** [comment]
I'm not clear on what you're asking. Can you rephrase the question?

**Comment by artman** [comment]
I think @kronik's asking how to use class inheritance with Interactors, Presenters and Routers, in order to share common code in super-classes.

**Comment by AttwellBrian** [comment]
> I think @kronik's asking how to use class inheritance with Interactors, Presenters and Routers, in order to share common code in super-classes.

In general I'd discourage the use of inheritence based polymophism inside RIBs and instead encourage composition (of other RIBs or custom views) to re-use functionality.

Composition tends to maintain better in the long run. And RIBs are designed to support it well. Plus, any RIB/MVC/VIPER pattern is extra hard to maintain for inheritance based polymophism because you end up needing to maintain multiple inheretence trees (interactor tree, router tree, etc).

**Comment by artman** [comment]
Closing as answered / stale.
